src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -162,12 +162,15 @@ struct PerfRecordSample {
   if (flags.sample_type & PERF_SAMPLE_STACK_USER) {
     ring_buffer->ReadRawAtOffset(&event.stack_size, current_offset, sizeof(uint64_t));
     current_offset += sizeof(uint64_t);
-    event.stack_data = make_unique_for_overwrite<uint8_t[]>(event.stack_size);
+    if (event.stack_size != 0u) {
+      ring_buffer->ReadRawAtOffset(
+          &event.dyn_size, current_offset + (event.stack_size * sizeof(uint8_t)), sizeof(uint64_t));
+    }
+    event.stack_data = make_unique_for_overwrite<uint8_t[]>(event.dyn_size);
     ring_buffer->ReadRawAtOffset(event.stack_data.get(), current_offset,
-                                 event.stack_size * sizeof(uint8_t));
+                                 event.dyn_size * sizeof(uint8_t));
```

Comment:
Shouldn't this be also in the `if`?

Reply:
Okay, I'll add those two lines inside the `if` but I think if stack_size is zero it won't make a difference.

Reply:
If stack_size is 0, we won't write the dyn_size value. But in the next line we access dyn_size. This would break if someone decided not to intiialize the struct anymore.

Reply:
Should be done!